web/src/main/java/com/marklogic/hub/web/web/CurrentProjectController.java

```diff
@@ -212,24 +217,33 @@ public String preInstallCheck() {
         return jsonResponse;
     }
 
-    private void startProjectWatcher() throws IOException {
-        String flowsDir = hubConfig.getFlowsDir().toString();
-        String stepDefinitionsDir = hubConfig.getStepDefinitionsDir().toString();
-        String entitiesDir = hubConfig.getHubEntitiesDir().toString();
-        String mappingsDir = hubConfig.getHubMappingsDir().toString();
+    private void startProjectWatcher() {
+        File flowsDir = hubConfig.getFlowsDir().toFile();
+        File stepDefinitionsDir = hubConfig.getStepDefinitionsDir().toFile();
+        File entitiesDir = hubConfig.getHubEntitiesDir().toFile();
+        File mappingsDir = hubConfig.getHubMappingsDir().toFile();
         if (!watcherService.hasListener(this)) {
-            watcherService.watch(flowsDir);
-            watcherService.watch(stepDefinitionsDir);
-            watcherService.watch(entitiesDir);
-            watcherService.watch(mappingsDir);
+            enableWatcherService(flowsDir);
+            enableWatcherService(stepDefinitionsDir);
+            enableWatcherService(entitiesDir);
+            enableWatcherService(mappingsDir);
             //watch ml-modules/root/custom-modules/ingestion|mapping|mastering dirs
             for (StepDefinition.StepDefinitionType stepType : StepDefinition.StepDefinitionType.values()) {
-                watcherService.watch(hubProject.getCustomModulesDir().resolve(stepType.toString().toLowerCase()).toString());
+                enableWatcherService(hubProject.getCustomModulesDir().resolve(stepType.toString().toLowerCase()).toFile());
             }
             watcherService.addListener(this);
         }
     }
 
+    private void enableWatcherService(File dir) {
+        try{
+            watcherService.watch(dir.toString());
+        }
+        catch (IOException ex){
+            logger.warn("Unable to enable watcher service for "+ dir.toString(), ex);
```

Comment:
I think for an exception like this - which is often true of warnings - we should just log the exception message, but not the full stacktrace. If we do the stacktrace for several exceptions, the user may thing something has gone horribly wrong when she sees several stacktraces being logged. 

So for this, and generally for warnings, I recommend doing the following:

    logger.warn("Unable to enable watcher server for directory: " + dir.toString() + "; cause: " + ex.getMessage());